lib/ood_core/job/adapters/linux_host/templates/script_wrapper.erb.sh

```diff
@@ -1,5 +1,20 @@
 #!/bin/bash
-hostname
+hostnames=$(hostname -A)
+hostnameFound=false
+for host in $SSH_HOSTS
+do
+    if [[ " ${hostnames[@]} " =~ " ${host} " ]]; then
+        hostnameFound=true
+        hostname=$host
+    fi
+done
```

Comment:
I don't believe `SSH_HOSTS` is being set on the destination. I just got `[]` and I believe it's because it's evaluated on the other side, and we're not passing it through ssh with `SendEnv`, which may be impossible as sshd configures to allow only certain variables through `AcceptEnv`. 

I can't see what we accept because of permissions, but the point may be moot as you could just write out hosts as an environment variable inlined in the script, so you don't have to pass through ssh. 

Something like this, writing the ruby array out to a bash array:
`SSH_HOSTS=(<%= hosts.join(' ').to_s =>)`

Reply:
Oh yes, this is quite different than the other adapters. So that also means that if `OodCore::Job::Script#job_environment` has values set, we are not setting these when we start the processes on the remote host?

Reply:
It looks like we export them inline in the script, so we're not passing them through ssh.

This bit here gets written out as just `<%= environment %>`. 
https://github.com/OSC/ood_core/blob/9eaf3cfe937657104f7d16b4bf2452be18cbdebc/lib/ood_core/job/adapters/linux_host/launcher.rb#L184-L193